### ISSUE TITLE
L7Policies in service_builder.py retrieves recursive information about policy

### ISSUE THREAD

**Issue by pjbreaux** [F5Networks/f5-openstack-lbaasv2-driver#276]
@richbrowne: checkout this issue

* Details: The [call](https://github.com/F5Networks/f5-openstack-lbaasv2-driver/blob/feature.l7switching/f5lbaasdriver/v2/bigip/service_builder.py#L413) to get the policies based on a listener id retrieves information recursively about that policy, including the listener it is attached to and the pool for that listener etc... The call currently being made is to_dict(). After some testing, we found that we can cut down the information by doing the following:

to_dict(listener=False)
```
[{
	u 'redirect_pool_id': None,
	u 'description': u '',
	u 'admin_state_up': True,
	u 'rules': [],
	u 'tenant_id': u 'c493efb7607f44d7a6284ce39cc7ca58',
	u 'listener_id': u '0bdbc44d-b1e1-4da1-8a23-cd10adec1bff',
	u 'redirect_pool': None,
	u 'redirect_url': None,
	u 'action': u 'REJECT',
	u 'position': 1,
	u 'provisioning_status': u 'PENDING_CREATE',
	u 'id': u '14f2fd11-2c33-41bf-bf53-1a07191d90ab',
	u 'name': u ''
}]
```
In comparison, when doing to_api_dict(), we get:
```
[{
	u 'redirect_pool_id': None,
	u 'description': u '',
	u 'admin_state_up': True,
	u 'rules': [],
	u 'tenant_id': u '4f115f7848d84e3c8061dcfe370e0353',
	u 'listeners': [{
		u 'id': u 'af26d4cc-65cf-4b6e-ae52-3906c45f2614'
	}],
	u 'redirect_url': None,
	u 'action': u 'REJECT',
	u 'position': 1,
	u 'id': u '3447e9d8-b9ea-480f-835e-a45ce4bd0c9d',
	u 'name': u ''
}]
```

#### Agent Version
This was found on branch feature.l7switching

**Comment by richbrowne** [comment]
Add the following attribute filters:


L7Policy:

************ listener=False, and you might also have to delete
redirect_pool, and add redirect_pool_id.

    def to_api_dict(self):
        ret_dict = super(L7Policy, self).to_dict(
            listener=False, listener_id=False,
            provisioning_status=False, redirect_pool=False)
        ret_dict['listeners'] = []
        if self.listener:
            ret_dict['listeners'].append({'id': self.listener.id})
        ret_dict['rules'] = [{'id': rule.id} for rule in self.rules]
        return ret_dict


L7Rule:

********* set policy to False, l7_policy_id stays.

   def to_api_dict(self):
        ret_dict = super(L7Rule, self).to_dict(
            provisioning_status=False,
            policy=False, l7policy_id=False)
        ret_dict['policies'] = []
        if self.policy:
            ret_dict['policies'].append({'id': self.policy.id})
        return ret_dict

**Comment by pjbreaux** [comment]
Fixed by PR #280
